modules/View.js

```diff
@@ -698,22 +721,79 @@
      * @method invokeDetached
      */
     invokeDetached: function() {
-      // No need to check if child views are actually detached, because if parent is detached, children are detached.
       if (this.__attachedCallbackInvoked) {
         this._detached();
         this.__attachedCallbackInvoked = false;
       }
-      _.each(this.__childViews, function(view) {
-        view.invokeDetached();
-      });
-      _.each(this.__sharedViews, function(view) {
-        view.invokeDetached();
+      _.each(this.getTrackedViews(), function(view) {
+        // No need to check if tracked view is actually detached from the DOM as long as they are attached to the parent and the if parent is detached,
+        // then the tracked view is detached.
+        if (view.isAttachedToParent()) {
+          view.invokeDetached();
+        }
       });
     },
 
     /************** Private methods **************/
 
     /**
+     * Gets the hash from id to views of the correct views given the options.
+     * @param [options={}] {Object}  Optional options.
+     *   @param [options.shared=false] {Boolean} The view is a shared view instead of a child view
+     *                                           (shared views are not disposed when the parent is disposed)
+     * @method __getTrackedViewsHash
+     * @return the object map cotaining either the shared or child views keyed by their cid
+     * @private
+     */
+    __getTrackedViewsHash: function(options) {
+      options = options || {};
+      if (options.shared) {
+        return this.__sharedViews;
+      } else {
+        return this.__childViews;
+      }
+    },
+
+    /**
+     * Used internally by Torso.View to keep a cache of tracked views and their current injection sites before detaching during render logic.
+     * @private
+     * @method __updateInjectionSiteMap
+     */
+    __updateInjectionSiteMap: function() {
+      var parentView = this;
+      this.__injectionSiteMap = {};
+      _.each(this.getTrackedViews(), function(view) {
+        if (view.isAttachedToParent() && view.injectionSite) {
+          parentView.__injectionSiteMap[view.injectionSite.attr('inject')] = view;
```

Comment:
Hmm... this made me re-look into it. this.injectView relies on defining "inject" attr. this.attach which is used by this.attachView does not rely on it.

If people use attachView and not injectView, which I guess is possible even if that's not what I do, then the "inject" attr will not be defined.

The outcome of this use case is that if you use attachView and don't use a standard injection site, and then call transitionNewViewIntoSite with an injection site name and want/hope that it will automatically find the old view somehow to transition it out, that view will not be found. Which makes sense that it wouldn't find it, because come on, transitionNewViewIntoSite is looking for the same transition site, and you didn't even give one to the first child view.

And on top of all that, if you did want to transition out a view that wasn't given a "proper" injection site, you can always specifiy it as the option "previousView" when calling transitionNewViewIntoSite.

I think we're okay